tests/lib/rules/use-isnan.js

```diff
@@ -349,140 +349,367 @@ ruleTester.run("use-isnan", rule, {
+                    messageId: "replaceWithIsNaN",
+                    output: '!Number.isNaN("abc");'
+                }]
+            }]
         },
         {
             code: "Number.NaN <= \"abc\";",
-            errors: [comparisonError]
+            errors: [{
+                ...comparisonError,
+                suggestions: [{
+                    messageId: "replaceWithIsNaN",
+                    output: '!Number.isNaN("abc");'
+                }]
+            }]
         },
         {
             code: "\"abc\" <= Number.NaN;",
-            errors: [comparisonError]
+            errors: [{
+                ...comparisonError,
+                suggestions: [{
+                    messageId: "replaceWithIsNaN",
+                    output: 'Number.isNaN("abc");'
+                }]
+            }]
         },
         {
             code: "Number.NaN >= \"abc\";",
-            errors: [comparisonError]
+            errors: [{
+                ...comparisonError,
+                suggestions: [{
+                    messageId: "replaceWithIsNaN",
+                    output: 'Number.isNaN("abc");'
+                }]
+            }]
         },
         {
             code: "\"abc\" >= Number.NaN;",
-            errors: [comparisonError]
+            errors: [{
+                ...comparisonError,
+                suggestions: [{
+                    messageId: "replaceWithIsNaN",
+                    output: '!Number.isNaN("abc");'
+                }]
+            }]
         },
         {
             code: "x === Number?.NaN;",
             languageOptions: { ecmaVersion: 2020 },
-            errors: [comparisonError]
+            errors: [{
+                ...comparisonError,
+                suggestions: [{
+                    messageId: "replaceWithIsNaN",
+                    output: "Number.isNaN(x);"
+                }]
+            }]
+        },
+        {
+            code: "x !== Number?.NaN;",
+            languageOptions: { ecmaVersion: 2020 },
+            errors: [{
+                ...comparisonError,
+                suggestions: [{
+                    messageId: "replaceWithIsNaN",
+                    output: "!Number.isNaN(x);"
+                }]
+            }]
         },
         {
             code: "x === Number['NaN'];",
-            errors: [comparisonError]
+            errors: [{
+                ...comparisonError,
+                suggestions: [{
+                    messageId: "replaceWithIsNaN",
+                    output: "Number.isNaN(x);"
+                }]
+            }]
+        },
+        {
+            code: `/* just
+                adding */ x /* some */ === /* comments */ NaN; // here`,
+            errors: [{
+                ...comparisonError,
+                suggestions: [{
+                    messageId: "replaceWithIsNaN",
+                    output: `/* just
+                adding */ Number.isNaN(x); // here`
+                }]
+            }]
```

Comment:
I think this looks okay to me.

Reply:
Is it OK to lose some comments when fixing?

Reply:
Good question -- ideally we don't lose any comments, but since this is a suggestion, I think it's okay because people will manually need to accept it and can always undo it if it's a problem.

Reply:
I've had a similar case in ts-eslint, and there we've decided to aggregate all of the comments and throw them somewhere above the code.

Code:
https://github.com/typescript-eslint/typescript-eslint/blob/01556f50376b87eff21c42d121b627e3a6f003f0/packages/eslint-plugin/src/rules/no-array-delete.ts#L89-L102

Test:
https://github.com/typescript-eslint/typescript-eslint/blob/01556f50376b87eff21c42d121b627e3a6f003f0/packages/eslint-plugin/tests/rules/no-array-delete.test.ts#L510-L547

Do you think it's worth doing the same here?

Reply:
I also think it's okay for suggestions to just remove comments that happen to be there. As @nzakas noted, if it's a problem in some cases, user can undo the change and manually fix the code.